docs/standard/microservices-architecture/multi-container-microservice-net-applications/background-tasks-with-ihostedservice.md

```diff
@@ -0,0 +1,260 @@
+{            
+    //Other DI registrations;
+
+    // Register Hosted Services
+    services.AddSingleton<IHostedService, GracePeriodManagerService>();
+    services.AddSingleton<IHostedService, MyHostedServiceB>();
+    services.AddSingleton<IHostedService, MyHostedServiceC>();
+    //...
+}
+```
+
+In that code, the `GracePeriodManagerService` hosted service is real code from the Ordering business microservice in eShopOnContainers, while the other two are just two additional samples.
+
+The `IHostedService` background task execution is coordinated with the lifetime of the application (host or microservice, for that matter). You register tasks when the application starts and you have the opportunity to do some graceful action or clean-up when the application is shutting down.
+
+Without using `IHostedService`, you could always start a background thread to run any task. The difference is precisely at the app’s shutdown time when that thread would simply be killed without having the opportunity to run graceful clean-up actions.
+
+
+## The IHostedService interface
+
+When you register an `IHostedService`, .NET Core will call the `StartAsync()` and `StopAsync()` methods of your `IHostedService` type during application start and stop respectively. Specifically, start is called after the server has started and `IApplicationLifetime.ApplicationStarted` is triggered.
+
+The `IHostedService` as defined in .NET Core, looks like the following.
+
+```csharp
+namespace Microsoft.Extensions.Hosting
+{
+    //
+    // Summary:
+    //     Defines methods for objects that are managed by the host.
+    public interface IHostedService
+    {
+        //
+        // Summary:
+        // Triggered when the application host is ready to start the service.
+        Task StartAsync(CancellationToken cancellationToken);
+        //
+        // Summary:
+        // Triggered when the application host is performing a graceful shutdown.
+        Task StopAsync(CancellationToken cancellationToken);
+    }
+}
+```
+As you can imagine, you can create multiple implementations of IHostedService and register them at the `ConfigureService()` method into the DI container, as shown previously. All those hosted services will be started and stopped along with the application/microservice.
+
+As a developer, you are responsible for handling the stopping action or your services when `StopAsync()` method is triggered by the host.
+
+## Implementing IHostedService with a custom hosted service class deriving from the BackgroundService base class
+
+You could go ahead and create you custom hosted service class from scratch and implement the `IHostedService`, as you need to do when using .NET Core 2.0. 
+
+However, since most background tasks will need pretty similar needs in regard to the cancellation tokens management and other tipical operations, .NET Core 2.1 will be providing a very convenient abstract base class you can derive from, named BackgroundService.
```

Comment:
"need pretty similar needs" is awkward. Consider "will have similar needs"

Reply:
👍